src/geom/cell_tet.C

```diff
@@ -206,20 +206,27 @@ void Tet::choose_diagonal() const
   // Check for uninitialized diagonal selection
   if (this->_diagonal_selection==INVALID_DIAG)
     {
-      Real diag_01_23 = (this->point(0) + this->point(1) - this->point(2) - this->point(3)).norm_sq();
-      Real diag_02_13 = (this->point(0) - this->point(1) + this->point(2) - this->point(3)).norm_sq();
-      Real diag_03_12 = (this->point(0) - this->point(1) - this->point(2) + this->point(3)).norm_sq();
-
-      this->_diagonal_selection=DIAG_02_13;
-
-      if (diag_01_23 < diag_02_13 || diag_03_12 < diag_02_13)
-        {
-          if (diag_01_23 < diag_03_12)
-            this->_diagonal_selection=DIAG_01_23;
-
-          else
-            this->_diagonal_selection=DIAG_03_12;
-        }
+      auto M = [&](const unsigned i, const unsigned j) -> Point
+               { return this->point(i) + this->point(j); };
```

Comment:
Is this missing a 1/2? Assuming you are computing midpoints here...

Reply:
We are. And we were before too. But we avoid the division by two for performance I imagine. Because that doesn't change the order.

Reply:
Resolving since not applicable anymore.